File: CF_4C_Registration_system.py
n_t_c = int(input())
freq = dict()
for i in range(n_t_c):
    us_name = input()
    if us_name in freq:
        print(f'{us_name}{freq[us_name]}')
        freq[us_name] += 1
    else:
        freq[us_name] = 1
        print('OK')
# check if present in dict, if not set freq = 1 and PRINT OK, if alr present in dict, PRINT NAME+{FREQ}

File: Week_1/CF_4C_Registration_system.cpp
#include<iostream>
#include<stdio.h>
#include<unordered_map>
#include<string>
using namespace std;

int main(){
    int n_t_c;
    unordered_map<string,int> d;
    string k = "OK";
    cin >> n_t_c;
    for (int i = 0; i<n_t_c; i++){
        string s;
        cin >> s;
        if (d.count(s)){
            cout << s + to_string(d[s])<< endl;
            d[s]++;
        }
        else {
            d[s] = 1;
            cout << k << endl;
        }
    }
}
//# check if present in dict, if not set freq = 1 and PRINT OK, if alr present in dict, PRINT NAME+{FREQ}